atlasdb-timelock-server/src/main/java/com/palantir/atlasdb/timelock/TimeLockServerLauncher.java

```diff
@@ -58,19 +58,35 @@ private static void registerResources(
             TimeLockServer serverImpl) {
         Map<String, TimeLockServices> clientToServices = createTimeLockServicesForClients(
                 serverImpl,
-                configuration.clients());
+                configuration.clients(),
+                environment);
 
         environment.jersey().register(HttpRemotingJerseyFeature.DEFAULT);
         environment.jersey().register(new TimeLockResource(clientToServices));
     }
 
     private static Map<String, TimeLockServices> createTimeLockServicesForClients(
             TimeLockServer serverImpl,
-            Set<String> clients) {
+            Set<String> clients,
+            Environment environment) {
         ImmutableMap.Builder<String, TimeLockServices> clientToServices = ImmutableMap.builder();
         for (String client : clients) {
-            clientToServices.put(client, serverImpl.createInvalidatingTimeLockServices(client));
+            TimeLockServices invalidatingTimeLockServices = serverImpl.createInvalidatingTimeLockServices(client);
+
+            registerHealthChecks(environment, client, invalidatingTimeLockServices);
+
+            clientToServices.put(client, invalidatingTimeLockServices);
         }
         return clientToServices.build();
     }
+
+    private static void registerHealthChecks(Environment environment, String client,
+            TimeLockServices invalidatingTimeLockServices) {
+        environment.healthChecks().register(
+                client + "-timestamp-service",
+                new TimestampServiceHealthCheck(invalidatingTimeLockServices.getTimestampService()));
+        environment.healthChecks().register(
+                client + "-lock-service",
+                new LockServiceHealthCheck(invalidatingTimeLockServices.getLockService()));
```

Comment:
Indeed, if a node has been partitioned and _thinks_ it's still the leader, this healthcheck could well succeed despite the service patently being unhealthy. (unless the act of grabbing the lock requires quorum?)